torch/csrc/jit/fusion_compiler.h

```diff
@@ -60,24 +60,37 @@ struct AnnotatedGraph {
   std::vector<TensorDesc> output_desc;
 };
 
-struct ConcatDesc {
-  size_t nSubtensors; // == 1 for outputs that are not concats, otherwise it is the number tensors concatenated
-  size_t dim; // dimension along which the concat occurs
+// Descriptor for chunk-ing an input tensor into subtensors
+// OR concat-ing an output tensor from subtensors
+struct PartitionDesc {
+  size_t nSubtensors; // == 1 for tensors that should not be operated on via chunk/cat
+  size_t dim; // dimension along which the chunk/concat occurs
   std::unique_ptr<TensorDesc> subtensorDesc; // descriptor for the subtensor, if it exists
-  ConcatDesc()
+  PartitionDesc()
   : nSubtensors(1), dim(0) {}
-  ConcatDesc(const TensorDesc & desc, size_t nSubtensors, size_t dim)
+  // PartitionDesc is used as metadata for either fused chunk or FusedConcat ops.
+  // Depending on which one it is used for, the contiguity calc is different.
+  PartitionDesc(const TensorDesc & desc, size_t nSubtensors, size_t dim, bool is_chunk)
   : nSubtensors(nSubtensors), dim(dim) {
     JIT_ASSERT(nSubtensors > 1);
     std::vector<bool> cont = desc.contiguity;
-    if(dim > 0) {
-      // when we narrow the concatenated output
-      // we make the size[dim] smaller while keeping the stride[dim] the same,
-      // meaning: stride[dim - 1] != stride[dim]*size[dim]
-      // so dim - 1 is no longer contiguous
-      cont[dim - 1] = false;
+    if (is_chunk) {
+      subtensorDesc.reset(new TensorDesc(desc.scalar_type, desc.contiguity));
```

Comment:
Are you sure that we don't need to modify the contiguity similarly to what we do below? Chunking along any dimension other than the first one will change a contiguous input to non-contiguous chunks.

```python
In [2]: import torch

In [3]: x = torch.randn(10, 10)

In [4]: x.chunk(10, 0)[0].is_contiguous()
Out[4]: True

In [5]: x.chunk(10, 1)[0].is_contiguous()
Out[5]: False
```

Reply:
Yes, `desc.contiguity` should already have the correct contiguity:
https://github.com/pytorch/pytorch/blob/60b58f153f0325eeb585d030da9af61f124b4aa7/torch/csrc/jit/fusion_compiler.cpp#L470-L476 
That part of the code constructs `desc` with the correct sizes / strides of each of the output chunks, and then the TensorDesc constructor computes the correct contiguity. `desc` is then passed into the PartitionDesc ctor. 

This behavior is different from what happens in a `cat` operation: for `aten::cat` PartitionDesc computes the correct contiguity, but for `aten::chunk` PartitionDesc is passed the correct contiguity. I'll add comments, a separate constructor, or move some code around as appropriate to make this distinction clearer.